### ISSUE TITLE
The Training is Stuck at First Epoch

### ISSUE THREAD

**Issue by jamalihuzaifa9** [matterport/Mask_RCNN#2655]
I am using 2 RTX 3080 Ti for training the model on custom data. I am using this [https://github.com/akTwelve/Mask_RCNN.git](url) repo which updated the Mask RCNN to tensorflow 2.x, the training is stuck at 1st epoch and doesn't produce any output.

**Comment by sajadtavakoli** [comment]
Hi, I am also struggling with this problem. I'm using tensorflow==1.10 and keras==2.0.8. when I run the code, the model is stuck at the first epoch.

**Comment by jamalihuzaifa9** [comment]
@sajadtavakoli I suggest you should use tensorflow>=1.13.0 and keras>=2.0.8 because that's the basic requirement for running the Mask RCNN model. You are using tensorflow==1.10 which is not supported.

**Comment by sajadtavakoli** [comment]
@jamalihuzaifa9 Thanks Jamal for your reply and attention.
My problem was solved. I found out that when I run the code in python console, the code is executed up to the first epoch and then stops (without any error). but when I run the code in the terminal, the training process is performed correctly. I have not used arg parser in my code.

**Comment by AndySung320** [comment]
I work with:
**RTX 3060
python 3.8
cuda_11.0.2_451.48_win10
cudnn-11.0-windows-x64-v8.0.4.30
TF: 2.3.0
Keras: 2.4.3**

the problem you mention may caused by cuda version

**Comment by jamalihuzaifa9** [comment]
@AndySung320 You working with the same updated repo for Mask RCNN as mentioned in my initial comment ? And if so, are you able to train the model with the config mentioned by you?

**Comment by AndySung320** [comment]
@jamalihuzaifa9 
No, I use https://github.com/leekunhee/Mask_RCNN for tf2.X
By the way, when using tf2.X for training, the performance seems to be better.
Also, this repo has a bug but easy to deal with. 
The bug is that we can't use multiple training step in the **train function**, the loss value will be weird. So you just finish one training process and then start another training strategy (ex. change the learning rate , training layer, or epoch).

**Comment by jamalihuzaifa9** [comment]
@AndySung320 
Tried your config still the same issue it is stuck on the first epoch. I tried it on google colab and my system as well. My system is running Ubuntu 20.0.4 btw not windows.

**Comment by AndySung320** [comment]
I use Windows system, maybe there are some compatibility issue while using Ubuntu

**Comment by van-flow** [comment]
> @jamalihuzaifa9
> No, I use https://github.com/leekunhee/Mask_RCNN for tf2.X
> By the way, when using tf2.X for training, the performance seems to be better.
> Also, this repo has a bug but easy to deal with.
> The bug is that we can't use multiple training step in the **train function**, the loss value will be weird. So you just finish one training process and then start another training strategy (ex. change the learning rate , training layer, or epoch).

Can you share your file? I tried running Leekunhee's version in TF2.x, on Colab. It does not seem to use GPU and does not get past first epoch. Thanks.

**Comment by sachinkmohan** [comment]
Having the same problem here! 😞 So annoying!
